### DISCUSSION TITLE
how to only upload an dir (out of onedrive root dir) to remote with the same dir name ?

### DISCUSSION THREAD

**Discussion by shnwang** [abraunegg/onedrive#1865] [Q&A] [answered]
My onedrive root dir is ~/OneDrive ,and now I have a dir ~/dir1 with some files in it . I want to just upload the dir to remote with the same dir name . I run like this :**_onedrive --syncdir ~/dir1 --synchronize --upload-only  --no-remote-delete_**
but all the files be upload to **remote root dir** , this is not what I desire. My goal is to upload ~/dir1 ro remote ~/dir with files.
So could onedrive assign an option that dir name will be upload to ?
local dirA -> remote dirB

**Comment by abraunegg** [comment, marked as answer]
@miroskv 

> My onedrive root dir is ~/OneDrive ,and now I have a dir ~/dir1 with some files in it . I want to just upload the dir to remote with the same dir name . I run like this :**_onedrive --syncdir ~/dir1 --synchronize --upload-only --no-remote-delete_** but all the files be upload to **remote root dir** , this is not what I desire. My goal is to upload ~/dir1 ro remote ~/dir with files. So could onedrive assign an option that dir name will be upload to ? local dirA -> remote dirB

You cant do this.

You can only upload data that is within in your configured 'sync_dir' - this is how the API works.

You must either:

1. Move your `~/dir1` to `~/OneDrive/dir1`

or 

1. Symbolically link `~/OneDrive/dir1` -> `~/dir1`

**Reply by shnwang** [reply]
Thanks . For solution 1, my dir1 and OneDrive root dir may not in the same disk ,OneDrive disk may not have enough disk space .For solution 2, I have try, make a link but onedrive can not find the symbolically dir .Additionally ,dir and the sympolically not in the same disk and they have different file system.

**Reply by shnwang** [reply]
One more question, could I specify the destination dir for upload ?

**Reply by abraunegg** [reply]
> Thanks . For solution 1, my dir1 and OneDrive root dir may not in the same disk ,OneDrive disk may not have enough disk space .For solution 2, I have try, make a link but onedrive can not find the symbolically dir .Additionally ,dir and the sympolically not in the same disk and they have different file system.

Then your symbolic link is incorrect. You need to test your setup with `--dry-run`.

> One more question, could I specify the destination dir for upload ?

This is clearly articulated in the documentation. Please read the documentation.

**Reply by shnwang** [reply]
thanks for your reply!
